drivers/i2c/i2c_sam0.c

```diff
@@ -247,7 +239,7 @@ static bool i2c_sam0_dma_write_start(const struct device *dev)
 	dma_cfg.channel_direction = MEMORY_TO_PERIPHERAL;
 	dma_cfg.source_data_size = 1;
 	dma_cfg.dest_data_size = 1;
-	dma_cfg.user_data = dev;
+	dma_cfg.user_data = data;
```

Comment:
Looks suspicious. We haven't modified `i2c_sam0_dma_write_done` callback. It still expects to receive pointer to dev, not pointer to dev->data.

Similar change has been done in `i2c_sam0_dma_read_start`.

Reply:
It was hard to tell, I thought user_data wasn't being used at all by i2c_sam0_dma_write_done

Reply:
I see, now.. should be fixed up.